content/languages/javascript/authoring.md

```diff
@@ -0,0 +1,283 @@
+
+This article is not a standalone tutorial on creating kata or translations. It's meant to be a complementary, JavaScript-specific part of a more general set of HOWTOs and guidelines related to [content authoring](/authoring/). If you are going to create a JavaScript translation, or a new JavaScript kata from scratch, please make yourself familiar with the aforementioned documents related to authoring in general first. 
+
+
+## General info
+
+Any technical information related to the JavaScript setup on Codewars can be found on the [JavaScript reference](/languages/javascript/) page (language versions, available libraries, and setup of the code runner).
+
+## Description
+
+JavaScript code blocks can be inserted with JavaScript-specific part in [sequential code blocks](/references/markdown/extensions/#sequential-code-blocks):
+
+~~~
+```javascript
+
+...your code here...
+
+```
+~~~
+
+JavaScript-specific paragraphs can be inserted with [language conditional rendering](/references/markdown/extensions/#conditional-rendering):
+
+```
+~~~if:javascript
+
+...text visible only for JavaScript description...
+
+~~~
+
+~~~if-not:javascript
+
+...text not visible in JavaScript description...
+
+~~~
+```
+
+## Tasks and Requirements
+
+Some concepts don't always translate well to or from JavaScript. Because of this, some constructs should be avoided and some translations just shouldn't be done.
+- Avoid returning different data types depending on the situation (_"Return the result, or the string 'Error' if no result can be found..."_). JavaScript is dynamically typed, which is not the case for some other languages. Returning `null` might be appropriate in some situations, but throwing an error might be better in others.
+
+_TODO: Finish this_
+
+## Tests
+
+### Mocha
+
+JavaScript kata should use [Mocha](https://mochajs.org/) to implement and execute tests.
+
+#### Test grouping functions
+
+To create and present test output, Mocha uses parameters of the `describe` and `it` functions:
+
+```javascript
+describe('Fixed tests', () => {
+  it('Odd numbers', () => {
+    // some assertions...
+  });
+
+  it('Even numbers', () => {
+    // some assertions...
+  });
+});
+
+describe('Random tests', () => {
+  it('Small inputs', () => {
+    // some assertions...
+  });
+
+  it('Large inputs', () => {
+    // some assertions...
+  });
+});
+```
+
+`describe` blocks can contain `describe` or `it` blocks, but `it` blocks can only contain assertions.
+
+#### Dynamically generated test cases
+
+It's possible to put functions decorated with `it` in a loop and use them as a construct similar to parameterized test cases known from other testing frameworks, for example:
+
+```javascript
+describe("Generated test cases", () => {
+  const testCases = generateTestCases();
+  testCases.forEach(([msg, input, expected]) => {
+    it(msg, () => {
+      const actual = userSolution(input);
+      assert.strictEqual(actual, expected);
```

Comment:
@hobovsky Now the problem here is, Chai hasn't been mentioned yet; and it's used here. Having it the other way around could confuse people with the `describe` and `it` though, so idk...

Reply:
I don't think it's a big deal in this particular point, the way how it is now is ok-ish to me. If you are really concerned with the fact that `assert` is used without being introduced, then see if it helps when you move `Chai` part before `Mocha` part. As far as I see, the paragraph on assertions does not mention `describe` or `it` in any way?